### PR TITLE
Mask Generation Task Guide

### PR DESCRIPTION
**Pull request by merveenoyan**
Added mask generation task guide.

### CODE REVIEW

File: docs/source/en/tasks/mask_generation.md
Comment (NielsRogge) [inline docs/source/en/tasks/mask_generation.md:1]:
```suggestion
<!--Copyright 2024 The HuggingFace Team. All rights reserved.
```

File: docs/source/en/tasks/mask_generation.md
Comment (NielsRogge) [inline docs/source/en/tasks/mask_generation.md:21]:
```suggestion
on labeled datasets and are limited to the classes they have seen during training; they return a set of masks and corresponding classes, 
```

File: docs/source/en/tasks/mask_generation.md
Comment (NielsRogge) [inline docs/source/en/tasks/mask_generation.md:27]:
```suggestion
that the prompt is pointing out to. The second one is the segment everything mode. In segment everything, given an 
```

File: docs/source/en/tasks/mask_generation.md
Comment (NielsRogge) [inline docs/source/en/tasks/mask_generation.md:56]:
```suggestion
The easiest way to infer mask generation models is to use the `mask-generation` pipeline.
```

File: docs/source/en/tasks/mask_generation.md
Comment (merveenoyan) [reply docs/source/en/tasks/mask_generation.md:20]: I'll fix this, it should've been semantic_segmentation so

**Review by MKhalusova — COMMENTED** [review]
Great work, @merveenoyan ! Left some suggestions.

File: docs/source/en/tasks/mask_generation.md
Comment (merveenoyan) [reply docs/source/en/tasks/mask_generation.md:24]: the model is actually manually labeled with masks, semi-automatic labeled, and fully automatically labeled, however the label here doesn't imply class masks like fully supervised image segmentation. will try to put this properly

File: docs/source/en/tasks/mask_generation.md
Comment (merveenoyan) [reply docs/source/en/tasks/mask_generation.md:25]: yes it is, will clarify

**Comment by merveenoyan** [comment]
@MKhalusova thanks, I addressed your comments I think

**Review by MKhalusova — APPROVED** [review]
Thanks for addressing the feedback, LGTM!

**Comment by MKhalusova** [comment]
@ArthurZucker please take a look. the guide LGTM

File: docs/source/en/tasks/mask_generation.md
Comment (khipp) [inline docs/source/en/tasks/mask_generation.md:22]:
```suggestion
Mask generation models are trained on large amounts of data and operate in two modes:
```

File: docs/source/en/tasks/mask_generation.md
Comment (khipp) [inline docs/source/en/tasks/mask_generation.md:81]:
```suggestion
The `masks` looks like the following:
```

File: docs/source/en/tasks/mask_generation.md
Comment (khipp) [inline docs/source/en/tasks/mask_generation.md:183]:
```suggestion
You can also do box prompting in a similar fashion to point prompting. You can simply pass the input box in the format of a list
```

File: docs/source/en/tasks/mask_generation.md
Comment (khipp) [inline docs/source/en/tasks/mask_generation.md:185]: You used "post-process" in line 138. I would keep it consistent and stick to either.

File: docs/source/en/tasks/mask_generation.md
Comment (ArthurZucker) [inline docs/source/en/tasks/mask_generation.md:75]: should we just not split this and leave it to the window manager?

File: docs/source/en/tasks/mask_generation.md
Comment (ArthurZucker) [inline docs/source/en/tasks/mask_generation.md:148]:
```suggestion

```python
input_points = [[[2592, 1728]]] # point location of the bee

inputs = processor(image, input_points=input_points, return_tensors="pt").to(device)
with torch.no_grad():
    outputs = model(**inputs)
masks = processor.image_processor.post_process_masks(outputs.pred_masks.cpu(), inputs["original_sizes"].cpu(), inputs["reshaped_input_sizes"].cpu())
```
```

**Review by ArthurZucker — APPROVED** [review]
Thanks for the nice documentation!

File: docs/source/en/tasks/mask_generation.md
Comment (merveenoyan) [reply docs/source/en/tasks/mask_generation.md:22]: I would rather not have this.

File: docs/source/en/tasks/mask_generation.md
Comment (ArthurZucker) [reply docs/source/en/tasks/mask_generation.md:75]: No No i just meant that you have 4lines here
```suggestion
Let's segment everything. `points-per-batch` enables parallel inference of points in segment everything mode. This enables faster inference, but consumes more memory. Moreover, SAM only enables batching over points and not the images. `pred_iou_thresh` is the IoU confidence threshold where only the masks above that certain threshold are returned.
```

Comment (ArthurZucker) [reply docs/source/en/tasks/mask_generation.md:75]: 😈

File: docs/source/en/tasks/mask_generation.md
Comment (merveenoyan) [inline docs/source/en/tasks/mask_generation.md:182]:
```suggestion
to the model, then post-process the output again.
```

**Comment by merveenoyan** [comment]
@ArthurZucker can you merge this if it's ok?

File: docs/source/en/tasks/mask_generation.md
Comment (merveenoyan) [inline docs/source/en/tasks/mask_generation.md:146]:
```suggestion
```
```

**Pull request merged**
